src/Illuminate/Auth/Access/Gate.php

```diff
@@ -216,6 +216,32 @@ public function define($ability, $callback)
         return $this;
     }
 
+    /**
+     * Defines multiple abilities based on a class public methods and name.
+     *
+     * @param  string|class-string  $class
+     * @param  string|null  $as
+     * @return $this
+     */
+    public function defineWithClass($class, $as = null)
+    {
+        if (empty($as)) {
+            $as = Str::kebab(class_basename($class));
+        }
+
+        foreach ((new ReflectionClass($class))->getMethods() as $method) {
+            if ($method->isPublic()
+                && !$method->isStatic()
+                && !$method->isInternal()
+                && $method->getName() !== 'before'
+                && !Str::startsWith($method->getName(), '__')) {
+                $this->define(Str::kebab($method->getName()).'-'.$as, [$class, $method->getName()]);
```

Comment:
I can't recall off the top of my head: do Policies allowing auto-DI in the `__construct()` method? Either way, it might be beneficial to allow that for these defineWithClass methods as well?

```php suggestion
                $classInstance ??= app($class);
                $this->define(Str::kebab($method->getName()).'-'.$as, [$classInstance, $method->getName()]);
```